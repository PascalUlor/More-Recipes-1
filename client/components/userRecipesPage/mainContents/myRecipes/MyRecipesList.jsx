import React from 'react';
import PropTypes from 'prop-types';
import MyRecipe from './MyRecipe.jsx';

const MyRecipesList = ({ myRecipes }) => (
    <div className="card-deck mb-5 mt-3">
        {myRecipes.length === 0 ?
            <div className='display-4 text-center text-danger'>
                There are no available recipes to display
            </div> :
            myRecipes.map(myRecipe => <MyRecipe key={myRecipe.id} myRecipe={myRecipe}/>)
        }
    </div>
);

MyRecipesList.propTypes = {
    myRecipes: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array
  ]).isRequired
};

export default MyRecipesList;