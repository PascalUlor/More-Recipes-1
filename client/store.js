import { createStore, applyMiddleware, compose } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import indexReducer from './reducers/index';


const env = process.env.NODE_ENV || development;

let middleware = compose(applyMiddleware(thunk,
    reduxImmutableStateInvariant()), window.devToolsExtension ?
  window.devToolsExtension() : f => f);

if (env === 'production') {
  middleware = applyMiddleware(thunk);
}

const store = createStore(
  indexReducer,
  middleware
);

export default store;
