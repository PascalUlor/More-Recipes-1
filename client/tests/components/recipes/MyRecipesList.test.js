/* eslint-disable */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import store from '../../../store';
import MyRecipesList from
  '../../../components/userRecipesPage/mainContents/myRecipes/MyRecipesList.jsx';


const recipe = {
  title: 'new test title',
  ingredients: 'ing1, ing2, ingredient3, ingredient4, ingredient5, ingredient6',
  procedures: 'new test procedure for this new added recipe',
  imageFile: {},
  imageSrc: '/images/noimageyet.jpg',
  errors: {}
};
let props = {
  myRecipes: [{ ...recipe, id: 89 }, { ...recipe, id: 90 }],
  setCurrentRecipe: jest.fn(),
};
const setCurrentRecipeSpy = sinon.spy(props, 'setCurrentRecipe');
describe('Testing MyRecipesList component', () => {
  it('Should render without error', () => {
    const mountWrapper = mount((
      <Provider store={store}>
        <Router><MyRecipesList {...props}/></Router>
      </Provider>
    ));
    expect(toJson(mountWrapper)).toMatchSnapshot();
    const event = {};
    mountWrapper.find('button').at(0).simulate('click', event);
    expect(setCurrentRecipeSpy.called).toEqual(true);
    mountWrapper.find('button').at(1).simulate('click', event);
    expect(setCurrentRecipeSpy.called).toEqual(true);
  });

  it('Should render without error', () => {
    props = {
      ...props,
      myRecipes: [],
    };
    const mountWrapper = mount((
      <Provider store={store}>
        <Router><MyRecipesList {...props}/></Router>
      </Provider>
    ));
    expect(toJson(mountWrapper)).toMatchSnapshot();
  });
});
