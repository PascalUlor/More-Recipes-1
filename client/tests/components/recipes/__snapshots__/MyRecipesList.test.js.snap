// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MyRecipesList component Should render without error 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <MyRecipesList
        myRecipes={
          Array [
            Object {
              "errors": Object {},
              "id": 90,
              "imageFile": Object {},
              "imageSrc": "/images/noimageyet.jpg",
              "ingredients": "ing1, ing2, ingredient3, ingredient4, ingredient5, ingredient6",
              "procedures": "new test procedure for this new added recipe",
              "title": "new test title",
            },
            Object {
              "errors": Object {},
              "id": 89,
              "imageFile": Object {},
              "imageSrc": "/images/noimageyet.jpg",
              "ingredients": "ing1, ing2, ingredient3, ingredient4, ingredient5, ingredient6",
              "procedures": "new test procedure for this new added recipe",
              "title": "new test title",
            },
          ]
        }
        setCurrentRecipe={[MockFunction]}
      >
        <div
          className="card-deck mb-4 mt-4"
        >
          <MyRecipe
            key="90"
            myRecipe={
              Object {
                "errors": Object {},
                "id": 90,
                "imageFile": Object {},
                "imageSrc": "/images/noimageyet.jpg",
                "ingredients": "ing1, ing2, ingredient3, ingredient4, ingredient5, ingredient6",
                "procedures": "new test procedure for this new added recipe",
                "title": "new test title",
              }
            }
            setCurrentRecipe={[MockFunction]}
          >
            <div
              className="col-12 col-sm-6 col-md-6 col-lg-4 pt-2"
            >
              <div
                className="card"
              >
                <Link
                  replace={false}
                  to="/recipes/90/recipe-details"
                >
                  <a
                    href="/recipes/90/recipe-details"
                    onClick={[Function]}
                  >
                    <img
                      alt="Card image cap"
                      className="card-img-top img-fluid"
                    />
                  </a>
                </Link>
                <div
                  className="card-body text-center p-2"
                >
                  <div
                    className="row card-title"
                  >
                    <h6
                      className="col-9 mt-2 text-truncate"
                    >
                      new test title
                    </h6>
                    <small
                      className="col-3 mt-2"
                    >
                      <i
                        className="fa fa-eye"
                      />
                       
                    </small>
                  </div>
                  <div
                    className="card-text"
                  >
                    <small
                      className="text-success pr-1"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-thumbs-o-up"
                      />
                       
                    </small>
                    <button
                      className="btn btn-outline-danger btn-sm mr-1"
                      data-target="#deleteRecipeModal"
                      data-toggle="modal"
                      onClick={[Function]}
                    >
                      <small>
                        <i
                          className="fa fa-trash"
                        />
                         Delete
                      </small>
                    </button>
                    <button
                      className="btn btn-outline-success btn-sm"
                      data-target="#editRecipeModal"
                      data-toggle="modal"
                      onClick={[Function]}
                    >
                      <small>
                        <i
                          className="fa fa-edit"
                        />
                         Edit
                      </small>
                    </button>
                    <small
                      className="text-danger pl-1"
                    >
                      <i
                        className="fa fa-thumbs-o-down"
                      />
                       
                    </small>
                  </div>
                </div>
              </div>
            </div>
          </MyRecipe>
          <MyRecipe
            key="89"
            myRecipe={
              Object {
                "errors": Object {},
                "id": 89,
                "imageFile": Object {},
                "imageSrc": "/images/noimageyet.jpg",
                "ingredients": "ing1, ing2, ingredient3, ingredient4, ingredient5, ingredient6",
                "procedures": "new test procedure for this new added recipe",
                "title": "new test title",
              }
            }
            setCurrentRecipe={[MockFunction]}
          >
            <div
              className="col-12 col-sm-6 col-md-6 col-lg-4 pt-2"
            >
              <div
                className="card"
              >
                <Link
                  replace={false}
                  to="/recipes/89/recipe-details"
                >
                  <a
                    href="/recipes/89/recipe-details"
                    onClick={[Function]}
                  >
                    <img
                      alt="Card image cap"
                      className="card-img-top img-fluid"
                    />
                  </a>
                </Link>
                <div
                  className="card-body text-center p-2"
                >
                  <div
                    className="row card-title"
                  >
                    <h6
                      className="col-9 mt-2 text-truncate"
                    >
                      new test title
                    </h6>
                    <small
                      className="col-3 mt-2"
                    >
                      <i
                        className="fa fa-eye"
                      />
                       
                    </small>
                  </div>
                  <div
                    className="card-text"
                  >
                    <small
                      className="text-success pr-1"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-thumbs-o-up"
                      />
                       
                    </small>
                    <button
                      className="btn btn-outline-danger btn-sm mr-1"
                      data-target="#deleteRecipeModal"
                      data-toggle="modal"
                      onClick={[Function]}
                    >
                      <small>
                        <i
                          className="fa fa-trash"
                        />
                         Delete
                      </small>
                    </button>
                    <button
                      className="btn btn-outline-success btn-sm"
                      data-target="#editRecipeModal"
                      data-toggle="modal"
                      onClick={[Function]}
                    >
                      <small>
                        <i
                          className="fa fa-edit"
                        />
                         Edit
                      </small>
                    </button>
                    <small
                      className="text-danger pl-1"
                    >
                      <i
                        className="fa fa-thumbs-o-down"
                      />
                       
                    </small>
                  </div>
                </div>
              </div>
            </div>
          </MyRecipe>
        </div>
      </MyRecipesList>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Testing MyRecipesList component Should render without error 2`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <MyRecipesList
        myRecipes={Array []}
        setCurrentRecipe={
          [MockFunction] {
            "calls": Array [
              Array [
                90,
              ],
              Array [
                90,
              ],
            ],
          }
        }
      >
        <div
          className="not-found lead p-3 mt-3 mt-5"
        >
          <i
            className="fa fa-exclamation-triangle fa-3x pb-3 text-warning d-block"
          />
          You have no available recipes
        </div>
      </MyRecipesList>
    </Router>
  </BrowserRouter>
</Provider>
`;
